_posts/2023-09-05-use-opensearch-benchmark-with-randomized-query-terms.md

```diff
@@ -0,0 +1,171 @@
+
+## Prerequisites
+
+To set up to follow this example, you’ll need the following:
+
+- **Python 3.8+**: Follow instructions to install [Python 3.8+](https://www.python.org/downloads/)
+- **Java Development Kit (JDK) 17**: Follow instructions to [install JDK 17](https://www.oracle.com/java/technologies/downloads/)
+- **OpenSearch Benchmark**: Follow instructions to [install OpenSearch Benchmark](https://opensearch.org/docs/latest/benchmark/installing-benchmark/). I installed in a virtual environment in a scratch directory that also holds the custom workload and config. 
+- **OpenSearch Cluster**: For this walkthrough, I used [Docker Desktop](https://www.docker.com/products/docker-desktop/) to run OpenSearch on my machine. Follow the [OpenSearch Quick Start](https://opensearch.org/docs/latest/quickstart/) to create an OpenSearch cluster.
+- **Data in OpenSearch**: [BoardgameGeek provides APIs](https://boardgamegeek.com/wiki/page/BGG_XML_API2) that deliver data to you in Extensible Markup Language (XML). For this example, I used and extended [sample code from the Open Distro for Elasticsearch project that downloads some games from BoardGameGeek](https://github.com/opendistro-for-elasticsearch/sample-code/tree/main/BGG) to load games into OpenSearch in a `games` index. The data contains names, descriptions, ratings, user comments, and more. If you have your own data set, you can load that into OpenSearch instead.
+
+## Generate a custom workload
+
+Once you have your data loaded into the games index, create a working directory and `cd` into it. I created a directory `osb`. Create a custom workload by running:
+
+```opensearch-benchmark create-workload --workload=bgg --target-hosts="<your cluster endpoint>" --client-options="basic_auth_user:'<your user name>',basic_auth_password:'<your password>',verify_certs:false" --indices=games --output-path=.```
+
+**NOTE:** This example uses basic authentication. Be sure to replace `<your user name>`, and `<your user password>` with the values you used to set up the security plugin in the command-line above. Replace `<your cluster endpoint>` with the endpoint and port for your cluster.
+
+Opensearch benchmark creates a `bgg` directory with a number of files, including `games.json`, and `workload.py`. Examine `games.json`. This file contains the index settings that OpenSearch Benchmark will use when creating the games directory in your test cluster. In the settings section, you can see that Benchmark has created some templatized settings for `number_of_shards`, and `number_of_replicas`.
```

Comment:
If "Settings" appears as capitalized in the UI, it should be "In the Settings section".